tracing/src/lib.rs

```diff
@@ -174,6 +174,15 @@
 //! # fn main() {}
 //! ```
 //!
+//! For external functions which don't have built-in tracing support and can't have
+//! the `#[instrument]` attribute applied, the [`Span` struct][`Span`] has a
+//! [`in_scope()` function][`in_scope`] which can be used to easily wrap synchonous
+//! code in a span.
+//!
+//! For example:
+//! ```rust
+//! let json = info_span!("json.parse").in_scope(|| serde_json::from_slice(&buf))?;
```

Comment:
i think this example won't compile as-is. note that the variable `buf` is not declared, and we don't have a dev-dependency on `serde_json`. 

If we want to show `serde_json` as an example here, we should add a fake version of `serde_json::from_slice` so that the example compiles, and have a variable named `buf`. I think we may also need to add a `main` function returning `Result` so that the `?` operator is usable in the example:
```suggestion
//! # fn doc() -> Result<(), ()> {
//! # mod serde_json {
//! #    pub(crate) fn from_slice(buf: &[u8]) -> Result<(), ()> { Ok(()) }
//! # }
//! # let buf = &[];
//! let json = info_span!("json.parse").in_scope(|| serde_json::from_slice(&buf))?;
//! # let _ = json; // suppress unused variable warning
//! # Ok(())
//! # }
```